app/server/api/disbursement_api.py

```diff
@@ -0,0 +1,166 @@
+from decimal import Decimal
+
+from flask import Blueprint, request, make_response, jsonify, g
+from flask.views import MethodView
+from sqlalchemy import desc, asc
+from uuid import uuid4
+
+from server import db
+from server.schemas import credit_transfers_schema
+from server.models.user import User
+from server.models.disbursement import Disbursement
+from server.models.transfer_account import TransferAccount
+from server.models.credit_transfer import CreditTransfer
+from server.utils.auth import requires_auth
+from server.utils.transfer_filter import process_transfer_filters
+from server.utils.search import generate_search_query
+from server.utils.credit_transfer import make_payment_transfer
+from server.utils.transfer_enums import TransferSubTypeEnum, TransferModeEnum
+from server.models.utils import paginate_query, disbursement_credit_transfer_association_table
+from sqlalchemy.orm import lazyload, joinedload
+
+disbursement_blueprint = Blueprint('disbursement', __name__)
+
+class MakeDisbursementAPI(MethodView):
+
+    @requires_auth(allowed_roles={'ADMIN': 'admin'})
+    def post(self):
+        post_data = request.get_json()
+
+        # --- Handle Parameters ---
+
+        # HANDLE PARAM : search_stirng - Any search string. An empty string (or None) will just return everything!
+        search_string = post_data.get('search_string') or ''
+        # HANDLE PARAM : params - Standard filter object. Exact same as the ones Metrics uses!
+        encoded_filters = post_data.get('params')
+        filters = process_transfer_filters(encoded_filters)
+        # HANDLE PARAM : include_accounts - Explicitly include these users
+        include_accounts = post_data.get('include_accounts', [])
+        # HANDLE PARAM : include_accounts - Explicitly exclude these users
+        exclude_accounts = post_data.get('exclude_accounts', [])
+        disbursement_amount = abs(round(Decimal(post_data.get('disbursement_amount') or 0),6))
+
+        if include_accounts and exclude_accounts:
+            return { 'message': 'Please either include or exclude users (include is additive from the whole search, while exclude is subtractive)'}
+
+        order_arg = request.args.get('order') or 'DESC'
+        if order_arg.upper() not in ['ASC', 'DESC']:
+            return { 'message': 'Invalid order value \'{}\'. Please use \'ASC\' or \'DESC\''.format(order_arg)}
+        order = asc if order_arg.upper()=='ASC' else desc
+        sort_by_arg = request.args.get('sort_by') or 'rank'
+
+        # --- Build Disbursement Object ---
+        d = Disbursement()
+        d.creator_user = g.user
+        d.search_string = search_string
+        d.search_filter_params = encoded_filters
+        d.include_accounts = include_accounts
+        d.exclude_accounts = exclude_accounts
+
+        if include_accounts:
```

Comment:
The reason for "include_accounts" is to make the "uncheck all" in the frontend work efficiently. I'll give an example:  
Search is very sensitive, and relies more on sorting over exclusion. In fact, searching my full name on a live instance will give over 1000 results, even though my name isn't in the data set! As such, there'll be times when you want to enter a search term, hit a button to _uncheck_ all the results, and then explicitly check those to whom you want to disburse. Let's say you only want to pay <Transfer Account 14>. With `include_accounts` set up to not include results, your query looks like this: `include_accounts=[14]`. The other way, the frontend would have to load all of the search results and you'd have to do `exclude_accounts=[1...13, 15...1000]`

Reply:
yeah that did occur to me. the way i was thinking about it was that currently the search term is actually just superflous info - it's just going to be a superset of `include_accounts`.  Whereas under an addition/union rule, if account 14, then, u could actually just omit the search string (omit in that it returns no results, not that it returns all results) and return `include_accounts=[14]` on its own, because `set([14]) + set([]) = set([14])`

Reply:
The problem with omitting search string though, is that it would be divergent behaviour from the actual search API. Like with that, an empty string behaves much like the existing transfer accounts api (returns all). Could result in unexpected behaviour if you actually wanted to disburse to all